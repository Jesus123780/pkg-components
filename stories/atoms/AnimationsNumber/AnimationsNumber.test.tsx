import { render, screen } from '@testing-library/react'
import { AnimationsNumber } from './index'

describe('AnimationsNumber', () => {
  it('should render the default text "0.00" when no props are provided', () => {
    render(<AnimationsNumber />)
    const spans = screen.getAllByText(/[\d.]/)
    expect(spans).toHaveLength(4)
    expect(spans.map((el) => el.textContent).join('')).toBe('0.00')
  })

  it('should render each character of provided text as individual span', () => {
    const text = '123.45'
    render(<AnimationsNumber text={text} />)
    const spans = screen.getAllByText(/[\d.]/)
    expect(spans).toHaveLength(text.length)
    expect(spans.map((el) => el.textContent).join('')).toBe(text)
  })

  it('should match snapshot with custom text', () => {
    const text = '99.99'
    const { container } = render(<AnimationsNumber text={text} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should handle empty text gracefully', () => {
    render(<AnimationsNumber text="" />)
    const spans = screen.queryAllByText(/[\s\S]/) // Any character
    expect(spans.length).toBe(0)
  })
})
