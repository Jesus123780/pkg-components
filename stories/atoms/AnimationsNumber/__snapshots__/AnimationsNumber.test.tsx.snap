// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AnimationsNumber should match snapshot with custom text 1`] = `
<span
  class="sc-bdfCDU fnXUat"
>
  <span>
    9
  </span>
  <span>
    9
  </span>
  <span>
    .
  </span>
  <span>
    9
  </span>
  <span>
    9
  </span>
</span>
`;
